### ISSUE TITLE
Special Characters map does not seem to support TinyMCE charmap_append setting

### ISSUE THREAD

**Issue by mrwweb** [WordPress/gutenberg#7942]
**Describe the bug**
_This is a TinyMCE regression from latest WordPress (currently 4.9.7)._

As described on [#34371 on trac](https://core.trac.wordpress.org/ticket/34371), TinyMCE supports a `charmap_append` setting to extend the Character Map characters. Here is an [example of the code required to add characters](https://plugins.trac.wordpress.org/browser/hawaiian-characters/trunk/hawaiian-characters.php). It does not appear to work in Gutenberg (which includes the Character Map in the Classic Editor block).

**To Reproduce**
1. Install the [Hawaiian Characters plugin](https://wordpress.org/plugins/hawaiian-characters/) which extends the character map
2. Insert a "Classic Editor" block
3. Open the Character Map

**Expected behavior**
The added characters are available in the character map palette. At present, they are missing.

Disabling Gutenberg and viewing the character map shows the expected characters.

**Screenshots**
Current Default Editor:
![image](https://user-images.githubusercontent.com/1238696/42661478-2a8b1be0-85e3-11e8-8449-bf9cb4160dc9.png)

Gutenberg:
![image](https://user-images.githubusercontent.com/1238696/42661528-501b97b8-85e3-11e8-82a8-755cf7c0825b.png)

**Desktop:**
- Win10
- Firefox 61.0.1
- WordPress 4.9.7
- Gutenberg 3.2

**Comment by danielbachhuber** [comment]
I took a quick debug of this issue. Your settings are being passed through to TinyMCE:

```
[27-Jul-2018 14:41:12 UTC] array (
  'plugins' => 'charmap,colorpicker,hr,lists,media,paste,tabfocus,textcolor,fullscreen,wordpress,wpautoresize,wpeditimage,wpemoji,wpgallery,wplink,wpdialogs,wptextpattern,wpview',
  'toolbar1' => 'formatselect,bold,italic,bullist,numlist,blockquote,alignleft,aligncenter,alignright,link,unlink,wp_more,spellchecker,kitchensink',
  'toolbar2' => 'strikethrough,hr,forecolor,pastetext,removeformat,charmap,outdent,indent,undo,redo,wp_help',
  'toolbar3' => '',
  'toolbar4' => '',
  'external_plugins' => 
  array (
  ),
  'charmap_append' => '[["0256","A - kahako"],["0257","a - kahako"],["0274","E - kahako"],["0275","e - kahako"],["0298","I - kahako"],["0299","i - kahako"],["0332","O - kahako"],["0333","o - kahako"],["0362","U - kahako"],["0363","u - kahako"],["0699","okina"]]',
)
```

Maybe this is a bug with TinyMCE?

**Comment by youknowriad** [comment]
closed by #10968

**Comment by mrwweb** [comment]
Sorry for not getting to this sooner, @danielbachhuber, but it does seem to be working as of beta5!
